src/inert.js

```diff
@@ -18,638 +18,322 @@
-    if (element.localName == 'slot') {
-      const slot = /** @type {HTMLSlotElement} */ (element);
-      // Verify if ShadowDom v1 is supported.
-      const distributedNodes = slot.assignedNodes ?
-        slot.assignedNodes({flatten: true}) : [];
-      for (let i = 0; i < distributedNodes.length; i++) {
-        composedTreeWalk(distributedNodes[i], callback, shadowRootAncestor);
-      }
+  function addInertStyle(node) {
+    if (node.querySelector('style#inert-style')) {
       return;
     }
+    const style = document.createElement('style');
+    style.setAttribute('id', 'inert-style');
+    style.textContent = `
+    [inert], [inert] * {
+      pointer-events: none;
+      cursor: default;
+      user-select: none;
+      -webkit-user-select: none;
+      -moz-user-select: none;
+      -ms-user-select: none;
+    }`;
+    node.appendChild(style);
   }
 
-  // If it is neither the parent of a ShadowRoot, a <content> element, a <slot>
-  // element, nor a <shadow> element recurse normally.
-  let child = node.firstChild;
-  while (child != null) {
-    composedTreeWalk(child, callback, shadowRootAncestor);
-    child = child.nextSibling;
-  }
-}
-
-/**
- * Adds a style element to the node containing the inert specific styles
- * @param {Node} node
- */
-function addInertStyle(node) {
-  if (node.querySelector('style#inert-style')) {
-    return;
-  }
-  const style = document.createElement('style');
-  style.setAttribute('id', 'inert-style');
-  style.textContent = '\n'+
-                      '[inert] {\n' +
-                      '  pointer-events: none;\n' +
-                      '  cursor: default;\n' +
-                      '}\n' +
-                      '\n' +
-                      '[inert], [inert] * {\n' +
-                      '  user-select: none;\n' +
-                      '  -webkit-user-select: none;\n' +
-                      '  -moz-user-select: none;\n' +
-                      '  -ms-user-select: none;\n' +
-                      '}\n';
-  node.appendChild(style);
-}
-
-const inertManager = new InertManager(document);
-
-Object.defineProperty(Element.prototype, 'inert', {
-                        enumerable: true,
-                        get: function() {
-                          return this.hasAttribute('inert');
-                        },
-                        set: function(inert) {
-                          inertManager.setInert(this, inert);
-                        },
-                      });
+  const inertManager = new InertManager(document);
+
+  Object.defineProperty(Element.prototype, 'inert', {
+    enumerable: true,
+    get: function() {
+      return this.hasAttribute('inert');
+    },
+    set: function(inert) {
+      inertManager.setInert(this, inert);
+    },
+  });
+
+  const nativeFocus = HTMLElement.prototype.focus;
+  HTMLElement.prototype.focus = function() {
+    // If it is inert or into an inert node, no focus!
+    let target = this;
+    while (target && !target.inert) {
```

Comment:
So this checks for inertness when focus is attempted? If we're tabbing through a document, and traverse into an inert subtree, how do we know to skip over it and look for the next focusable element?